### DISCUSSION TITLE
How to run DocumentServer from the source code

### DISCUSSION THREAD

**Discussion by vv0rk0ffice** [ONLYOFFICE/DocumentServer#2949] [Q&A]
is there a wiki or guild to run `DocumentServer` from source code ??

**Comment by igwyd** [comment]
Hello @vv0rk0ffice if I understand your question correctly, then the build from source code is provided using the build_tools in this repository https://github.com/ONLYOFFICE/build_tools

**Comment by vv0rk0ffice** [comment]
thx for replying @igwyd 

I want to run this server in dev mode and tweak some stuff in webpp for example 

I have found  [this](https://github.com/ONLYOFFICE/build_tools/tree/master/develop) to run the server with docker and i can mount 

- sdkjs repo is located [here](https://github.com/ONLYOFFICE/sdkjs/)
- web-apps repo is located [here](https://github.com/ONLYOFFICE/web-apps/)
- server repo is located [here](https://github.com/ONLYOFFICE/server/)


so i can not edit the [core](https://github.com/ONLYOFFICE/core), [core-fonts](https://github.com/ONLYOFFICE/core-fonts) or [dictionaries](https://github.com/ONLYOFFICE/dictionaries)

**Comment by igwyd** [comment]
According to information from the developers you can add [core-fonts](https://github.com/ONLYOFFICE/core-fonts) and [dictionaries](https://github.com/ONLYOFFICE/dictionaries) similarly as sdkjs, web-apps, server. It is more difficult to add the core from the source code because it is C++, try to compile core in build_tools and replace it in docker.

**Comment by vv0rk0ffice** [comment]
ok thx @igwyd 

i will try it
